bigquery/user_defined_functions/string-functions.sql

```diff
@@ -14,16 +14,61 @@ CREATE OR REPLACE FUNCTION
+        OR LOWER(client_user_agent) LIKE "%batch%"
+        OR LOWER(client_user_agent) LIKE "%verification%"
+        OR LOWER(client_user_agent) LIKE "%qwantify%"
+        OR LOWER(client_user_agent) LIKE "%nuclei%"
+        OR LOWER(client_user_agent) LIKE "%ai%"
+        OR LOWER(client_user_agent) LIKE "%crawler%"
+        OR LOWER(client_user_agent) LIKE "%perl%"
+        OR LOWER(client_user_agent) LIKE "%java%"
+        OR LOWER(client_user_agent) LIKE "%test%"
+        OR LOWER(client_user_agent) LIKE "%scoop%"
+        OR LOWER(client_user_agent) LIKE "%fetch%"
+        OR LOWER(client_user_agent) LIKE "%adreview%"
+        OR LOWER(client_user_agent) LIKE "%cortex%"
+        OR LOWER(client_user_agent) LIKE "%nessus%"
+        OR LOWER(client_user_agent) LIKE "%bitdiscovery%"
+        OR LOWER(client_user_agent) LIKE "%postplanner%"
+        OR LOWER(client_user_agent) LIKE "%faraday%"
+        OR LOWER(client_user_agent) LIKE "%restsharp%"
+        OR LOWER(client_user_agent) LIKE "%hootsuite%"
+        OR LOWER(client_user_agent) LIKE "%mattermost%"
+        OR LOWER(client_user_agent) LIKE "%shortlink%"
+        OR LOWER(client_user_agent) LIKE "%retriever%"
+        OR LOWER(client_user_agent) LIKE "%auto%"
+        OR LOWER(client_user_agent) LIKE "%scrper%"
+        OR LOWER(client_user_agent) LIKE "%alyzer%"
+        OR LOWER(client_user_agent) LIKE "%dispatch%"
+        OR LOWER(client_user_agent) LIKE "%traackr%"
+        OR LOWER(client_user_agent) LIKE "%fiddler%"
+        OR LOWER(client_user_agent) LIKE "%crowsnest%"
+        OR LOWER(client_user_agent) LIKE "%gigablast%"
+        OR LOWER(client_user_agent) LIKE "%wakelet%"
+        OR LOWER(client_user_agent) LIKE "%installatron%"
+        OR LOWER(client_user_agent) LIKE "%intently%"
+        OR LOWER(client_user_agent) LIKE "%openurl%"
+        OR LOWER(client_user_agent) LIKE "%anthill%"
+        OR LOWER(client_user_agent) LIKE "%curb%"
+        OR LOWER(client_user_agent) LIKE "%trello%"
+        OR LOWER(client_user_agent) LIKE "%inject%"
+        OR LOWER(client_user_agent) LIKE "%ahc%" THEN "bot"
       WHEN LOWER(client_user_agent) LIKE "%mobile%"
         OR LOWER(client_user_agent) LIKE "%android%"
         OR LOWER(client_user_agent) LIKE "%whatsapp%"
         OR LOWER(client_user_agent) LIKE "%iphone%"
         OR LOWER(client_user_agent) LIKE "%ios%"
+        OR LOWER(client_user_agent) LIKE "%tablet%"
         OR LOWER(client_user_agent) LIKE "%samsung%" THEN "mobile"
       WHEN LOWER(client_user_agent) LIKE "%win%"
         OR LOWER(client_user_agent) LIKE "%mac%"
-        OR LOWER(client_user_agent) LIKE "%x11%" THEN "desktop"
+        OR LOWER(client_user_agent) LIKE "%x11%"
+        OR LOWER(client_user_agent) LIKE "%linux%"
+        OR LOWER(client_user_agent) LIKE "%opera%"
+        OR LOWER(client_user_agent) LIKE "%whatweb%"
+        OR LOWER(client_user_agent) LIKE "%microsoft%office%2014%" THEN "desktop"
```

Comment:
Someone's browsing the web on Word?

Reply:
My suspicion is that they're embedding a listing in a Word document...